### ISSUE TITLE
☂️ Remove accentColor and accentTextTheme from ThemeData et al.

### ISSUE THREAD

**Issue by HansMuller** [flutter/flutter#56918]
The [following proposal](https://docs.google.com/document/d/1kzIOQN4QYfVsc5lMZgy_A-FWGXBAJBMySGqZqsJytcE/edit#bookmark=id.on2uig6lmh3x) is from the "Deprecate/remove the accent properties" section of [flutter.dev/go/material-theme-system-updates](https://flutter.dev/go/material-theme-system-updates).

The Material Design spec no longer includes a color called ‘accent’. The ‘secondary’ color scheme color is roughly equivalent. The overall theme’s accent properties are rarely used within Flutter itself. For example the accentIconTheme is only used by FloatingActionButton and in that case, it’s only used (in one expression) for the sake of backwards compatibility. The accentColorBrightness property is only used by GridTile. We plan to replace component dependencies on  the accent properties with depencies on component themes, and the overall theme’s color scheme and text theme.

```dart
Color accentColor;
Brightness accentColorBrightness;
TextTheme accentTextTheme;
IconThemeData accentIconTheme;
```
The process for removing these properties will take a considerable amount of time because they’ve been around since early versions of Flutter.

1. Remove dependencies - but not definitions - from Flutter.
2. When all of the accent properties dependencies have been removed, send a breaking change announcement.
3. Deprecate the properties and constructor parameters.
4. Remove the deprecated properties and constructor parameters.

## `accentColor`

The following classes still depend on `accentColor` (checked items have been updated):

- [x] BottomNavigationBar PR: https://github.com/flutter/flutter/pull/76810
- [x] CheckboxListTile PR: #76908
- [x] ExpansionTile:  PR: https://github.com/flutter/flutter/pull/79829
Original PR was reverted: https://github.com/flutter/flutter/pull/77933
Added #78281 to enable migrating existing apps that depended on accentColor.
- [x] FloatingActionButton #78296
Some of this work was completed, see [flutter.dev/go/remove-fab-accent-theme-dependency](https://flutter.dev/go/remove-fab-accent-theme-dependency) and #46923.
- [x] GridTileBar PR: #78574
- [x] InputDecorator PR: #77999
- [x] ListTile PR: #77004
- [x] MaterialApp PR: #77652
- [x] ProgressIndicator PR: #77976
- [x] Added a ProgressIndicatorTheme. PR: [#81075](https://github.com/flutter/flutter/pull/81075)
- [x] PaginatedDataTable PR: #77744
- [x] RadioListTile PR: #76906
- [x] RefreshIndicator PR: #77884
- [x] Stepper PR: https://github.com/flutter/flutter/pull/77732
- [x] SwitchListTile PR: #76909
- [x] TabPageSelector PR: #77997

## `accentTextTheme`

The following classes still depend on `accentTextTheme` (checked items have been updated):

- [x] TimePicker PR: #78771
- [x] DayPicker PR: #78576
This class is deprecated.
- [x] Typography PR: #78371

**Comment by HansMuller** [comment]
The only remaining references to accentColor and accentTextTheme are for backwards compatibility in ThemeData, ColorScheme, and related tests.

**Comment by erdemyerebasmaz** [comment]
This is under ["Not yet released to stable"](https://flutter.dev/docs/release/breaking-changes#not-yet-released-to-stable) but InputDecorator PR: #77999 is on [stable ](https://github.com/flutter/flutter/commits/stable/packages/flutter/lib/src/material/input_decorator.dart)channel.
Other classes may be affected as well.

Was this intended?
